setup.sh

```
@@ -0,0 +1,385 @@
+            log_error "Unknown tool: $tool"
+            log_info "Available tools: ${AVAILABLE_TOOLS[*]}"
+            return 1
+        fi
+    done
+    return 0
+}
+
+# Check tool dependencies
+check_tool_deps() {
+    local tool=$1
+    local deps="$(get_tool_deps "$tool")"
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)